File: market/test/java/ru/yandex/market/wms/packing/controller/TaskControllerTest.java
package ru.yandex.market.wms.packing.controller;

import java.util.Collections;
import java.util.Map;

import com.github.springtestdbunit.annotation.DatabaseSetup;
import com.github.springtestdbunit.annotation.ExpectedDatabase;
import org.junit.jupiter.api.Test;

import ru.yandex.market.wms.packing.integration.PackingIntegrationTest;

import static com.github.springtestdbunit.annotation.DatabaseOperation.INSERT;
import static com.github.springtestdbunit.assertion.DatabaseAssertionMode.NON_STRICT_UNORDERED;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

class TaskControllerTest extends PackingIntegrationTest {

    @Test
    @DatabaseSetup(value = "/db/locations_setup_rov.xml", type = INSERT)
    @DatabaseSetup(value = "/db/base_setup.xml", type = INSERT)
    @ExpectedDatabase(value = "/db/integration/tasks/after-1.xml", assertionMode = NON_STRICT_UNORDERED)
    public void assignTasks() throws Exception {
        assertHttpCall(
                put("/tasks"),
                status().isOk(),
                "controller/tasks/assign-request.json"
        );
    }

    @Test
    @DatabaseSetup(value = "/db/locations_setup_rov.xml", type = INSERT)
    @DatabaseSetup(value = "/db/base_setup.xml", type = INSERT)
    @DatabaseSetup(value = "/db/dao/packing_task/get_tickets_multiple/setup.xml", type = INSERT)
    @DatabaseSetup(value = "/db/integration/tasks/setup.xml", type = INSERT)
    public void getAllTask() throws Exception {
        assertHttpCall(
                get("/tasks"),
                status().isOk(),
                Collections.emptyMap(),
                "controller/tasks/response/all.json"
        );

        assertHttpCall(
                get("/tasks"),
                status().isOk(),
                Map.of("users", "test-user,another-user"),
                "controller/tasks/response/available.json"
        );
    }
}

File: market/test/kotlin/ru/yandex/market/wms/consolidation/modules/consolidation/controller/TaskControllerTest.kt
package ru.yandex.market.wms.consolidation.modules.consolidation.controller

import com.github.springtestdbunit.annotation.DatabaseSetup
import com.github.springtestdbunit.annotation.ExpectedDatabase
import com.github.springtestdbunit.assertion.DatabaseAssertionMode
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.mock.mockito.SpyBean
import org.springframework.http.MediaType
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultMatchers
import ru.yandex.market.wms.common.spring.IntegrationTest
import ru.yandex.market.wms.common.spring.utils.FileContentUtils
import ru.yandex.market.wms.consolidation.modules.consolidation.dao.ConsolidationDao
import ru.yandex.market.wms.consolidation.modules.consolidation.dao.ConsolidationUserDao

class TaskControllerTest : IntegrationTest() {

    @SpyBean
    @Autowired
    private lateinit var consolidationDao: ConsolidationDao

    @SpyBean
    @Autowired
    private lateinit var consolidationUserDao: ConsolidationUserDao

    @Test
    @DatabaseSetup("/tasks/get/before.xml")
    fun getTasks() {
        mockMvc.perform(MockMvcRequestBuilders.get("/tasks/"))
            .andExpect(MockMvcResultMatchers.status().is2xxSuccessful)
            .andExpect(
                MockMvcResultMatchers.content()
                    .json(FileContentUtils.getFileContent("tasks/get/response.json"), false)
            )
    }

    @Test
    @DatabaseSetup("/tasks/put/before.xml")
    @ExpectedDatabase(
        value = "/tasks/put/after.xml",
        assertionMode = DatabaseAssertionMode.NON_STRICT_UNORDERED
    )
    fun assignTasks() {
        mockMvc.perform(
            MockMvcRequestBuilders.put("/tasks/")
                .contentType(MediaType.APPLICATION_JSON)
                .content(FileContentUtils.getFileContent("tasks/put/request.json"))
        )
            .andExpect(MockMvcResultMatchers.status().is2xxSuccessful)
            .andReturn()
    }

    @Test
    @DatabaseSetup("/tasks/put/before-error.xml")
    @ExpectedDatabase(
        value = "/tasks/put/after-error.xml",
        assertionMode = DatabaseAssertionMode.NON_STRICT_UNORDERED
    )
    fun assignTasksWhenUserIsOnDifferentPutWall() {
        mockMvc.perform(
            MockMvcRequestBuilders.put("/tasks/")
                .contentType(MediaType.APPLICATION_JSON)
                .content(FileContentUtils.getFileContent("tasks/put/request.json"))
        )
            .andExpect(MockMvcResultMatchers.status().is4xxClientError)
            .andReturn()
    }

    @Test
    @DatabaseSetup("/tasks/before.xml")
    fun assignAndGetTasks() {
        mockMvc.perform(
            MockMvcRequestBuilders.put("/tasks/")
                .contentType(MediaType.APPLICATION_JSON)
                .content(FileContentUtils.getFileContent("tasks/request.json"))
        )
            .andExpect(MockMvcResultMatchers.status().is4xxClientError)
            .andReturn()
        mockMvc.perform(MockMvcRequestBuilders.get("/tasks?users=sof-test,sof-test3"))
            .andExpect(MockMvcResultMatchers.status().is2xxSuccessful)
            .andExpect(
                MockMvcResultMatchers.content()
                    .json(FileContentUtils.getFileContent("tasks/response.json"), false)
            )
    }

}
